doc/doc_guide.rst

```diff
@@ -0,0 +1,201 @@
++-----------------------+-----------------------+-----------------------+
+| @p                    | Display a single word | ``@p transA``         |
+|                       | using a typewriter    |                       |
+|                       | font. Equivalent to   |                       |
+|                       | ``@c``                |                       |
++-----------------------+-----------------------+-----------------------+
+| @em                   | Display a single word | ``@em x``             |
+|                       | in italics.           |                       |
++-----------------------+-----------------------+-----------------------+
+
+.. note::
+
+    Not all Doxygen special commands are supported in Exhale, e.g., `grouping`_.
+    Some commands like `@name`_ could lead to errors when parsing overloaded functions.
+    To free yourself from debugging the Doxygen comments for hours, we recommend you only using the
+    above commands.
+
+| **How to include math formulas in Doxygen?**
+| Doxygen supports LaTeX math formulas in the documation. To include an inline formula that appears
+  in the running text, we need wrap it by a pair of ``@f$`` commands, for example:
+
+.. code:: none
+
+    Default is no smoothing, @f$\alpha = 0 @f$.
+
+This will result in: Default is no smoothing, |formula1|.
+
+For the longer formulas which are in seperate lines, we can put ``\f[`` and ``\f]`` commands between
+the formulas, for instance:
+
+.. code:: none
+
+    @f[
+       \operatorname{gelu}(x) = x \cdot \Phi(x)
+         = x \cdot \frac{1}{2}\left[
+            1 + \operatorname{erf}\left(\frac{x}{\sqrt{2}}\right)
+         \right]
+         \sim \operatorname{swish}(x, 1.702)
+    @f]
+
+This will result in:
+
+.. figure:: images/gelu_formula.png
+   :alt: Example of formula 2
+
+   Example of formula 2
+
+.. note::
+
+    Make sure the formula contains *valid* commands in `LaTeX’s math-mode`_.
+
+| **What are the recommendations for Doxygen comments?**
+| First of all, add Doxygen comments in the header files. You can find the examples of Doxygen
+  comments in `src/graph/expression_graph.h`_.  A good practice is to keep Doxygen comments as
+  intuitive and short as possible. Try not to introduce unnecessary vertical space (e.g., an empty
+  line). A basic template of Doxygen comments is shown as follows:
+
+.. code:: cpp
+
+    /**
+     * Brief summary.
+     * Detailed description. More detail.
+     * @see Some reference
+     * @param <name> Parameter description.
+     * @return Return value description.
+     */
+
+
+Documentation with Sphinx
+`````````````````````````
+
+Sphinx supports `Markdown`_ and `reStructuredText`_ documents. Our handwritten documentations are
+located in `doc`_.  The default format of Sphinx is `reStructuredText`_ and most of the framework's
+power comes from the richness of its default `reStructuredText`_ markup format.
+
+Markdown
+~~~~~~~~
+
+(in progress...)
```

Comment:
These two missing parts are the main reason why this PR is a draft?

Reply:
Exactly. I am still working on it. I am quite new to restructureText, so it took longer than I expected. Hopefully I will complete those parts by this week.

Reply:
Thanks. I don't think we need to write much in those sections. Probably a few sentences and a couple of links to external resources will be more than enough.

Reply:
I guess it is now ready to review. I highlighted several syntaxes that are newly added to myst-parser. The reStructuredText is a bit complex to explain in a few sentence. So I leave it to Exhale and Sphinx's docs. Feel free to comment or commit changes.